### PR TITLE
fix(text): use internal invariant, do not throw on production

### PR DESCRIPTION
**Pull request by adnasa**

### CODE REVIEW

File: packages/utils/src/invariant.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Strip out error messages for production
+// > Not providing an inline default argument for message as the result is smaller
+const invariant = (condition: any, message?: string): asserts condition => {
+  if (condition) {
+    return;
+  }
+  // GIVEN product, supress the message.
+  if (isProduction) {
+    return;
+  }
```
Comment (adnasa) [inline packages/utils/src/invariant.ts:15]: the original invariant throws on production
https://github.com/alexreardon/tiny-invariant/blob/master/src/tiny-invariant.ts#L18

Here we skip it.

File: packages/utils/src/invariant.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const isProduction: boolean = process.env.NODE_ENV === 'production';
+// @TODO: allow consumer to set the prefix e.g `[ Text.Body ]`
+const prefix: string = 'Invariant failed';
+
+// Throw an error if the condition fails
+// Strip out error messages for production
+// > Not providing an inline default argument for message as the result is smaller
+const invariant = (condition: any, message?: string): asserts condition => {
```
Comment (emmenko) [inline packages/utils/src/invariant.ts:8]: Ah nice, using `asserts`

File: packages/utils/src/invariant.ts (hunk 1 above)
Comment (emmenko) [inline packages/utils/src/invariant.ts:15]: Maybe we should log this instead of just ignoring it?

File: .changeset/rich-pandas-collect.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@commercetools-uikit/text': patch
+'@commercetools-uikit/utils': patch
```
Comment (emmenko) [inline .changeset/rich-pandas-collect.md:3]: Shouldn't we migrate all packages to use this? 🤔

Comment (tdeekens) [reply .changeset/rich-pandas-collect.md:3]: Can also be a separate step and PR once this is in.

File: packages/components/text/src/text.spec.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
-import invariant from 'tiny-invariant';
+import { invariant } from '@commercetools-uikit/utils';
 import { screen, render } from '../../../../test/test-utils';
 import Text from './text';
 
-jest.mock('tiny-invariant');
+jest.mock('@commercetools-uikit/utils', () => ({
+  ...jest.requireActual('@commercetools-uikit/utils'),
+  invariant: jest.fn(),
```
Comment (tdeekens) [inline packages/components/text/src/text.spec.js:8]: Now or later: the `invariant` could have a `__mock__`. Then you could just to `jest.mock()` maybe.

File: packages/utils/src/invariant.ts (hunk 1 above)
Comment (tdeekens) [reply packages/utils/src/invariant.ts:15]: Let's maybe ignore it for now and think about it. We could report to the console or sentry somehow.

**Review by tdeekens — APPROVED** [review]

File: .changeset/rich-pandas-collect.md (hunk 3 above)
Comment (emmenko) [reply .changeset/rich-pandas-collect.md:3]: Why migrate only the `Text` component then?

File: packages/utils/src/invariant.ts (hunk 1 above)
Comment (emmenko) [reply packages/utils/src/invariant.ts:15]: Why do you want to ignore it? I'm not sure I understand the intent here...

File: .changeset/rich-pandas-collect.md (hunk 3 above)
Comment (adnasa) [reply .changeset/rich-pandas-collect.md:3]: yes.
Let's just focus on `Text` test and `invariant`, then we have address the basics before migrating everything :)

File: packages/utils/src/invariant.ts (hunk 1 above)
Comment (adnasa) [reply packages/utils/src/invariant.ts:15]: > I think it's okay to first silence the error to fix the actual issue.

agree. let's get this out first, then we can refine it.

Comment (emmenko) [reply packages/utils/src/invariant.ts:15]: I see, I was missing the context.

File: packages/components/text/src/text.spec.js (hunk 4 above)
Comment (adnasa) [reply packages/components/text/src/text.spec.js:8]: I'll leave this as homework

**Comment by adnasa** [comment]
Okay.
this is again ready for review.
once this is approved, we merge it, but not **release** it until we migrate the other packages (separate PR).
I'll start on that now

**Review by tdeekens — APPROVED** [review]

**Review by emmenko — APPROVED** [review]

**Pull request merged**